pkg/endpoint/public_ip.go

```diff
@@ -88,6 +96,47 @@ func getPublicIP(submSpec *types.SubmarinerSpecification, k8sClient kubernetes.I
 	return "", nil
 }
 
+func resolveIPInAirGappedDeployment(k8sClient kubernetes.Interface, namespace, config string, airGapped bool) (string, error) {
+	if !airGapped {
+		return "", errors.Errorf("this function should be called only for air-gapped deployments")
+	}
```

Comment:
Since the function name indicates air gapped, we don't really need to pass in a flag. Plus the lone call site already checks for air gapped prior to calling.

```suggestion
func resolveIPInAirGappedDeployment(k8sClient kubernetes.Interface, namespace, config string) (string, error) {
```